qcodes/utils/dataset/doNd.py

```diff
@@ -483,6 +483,180 @@ def do2d(
+        else:
+            flat_setpoints = [()]
+        return flat_setpoints
+
+    def _find_parameters(*params: Union[_BaseParameter, int, float, ParamMeasT]
+                         ) -> Tuple[List[Union[_BaseParameter, ParamMeasT]],
+                                    Dict[Union[_BaseParameter, ParamMeasT], List[Union[int, float]]]]:
+        _params: List[Union[_BaseParameter, ParamMeasT]] = []
+        args_per_parameter: Dict[Union[_BaseParameter, ParamMeasT], List[Union[int, float]]] = {}
+        for par in params:
+            if isinstance(par, _BaseParameter) or callable(par):  # A QCodes parameter:
+                _params.append(par)
+                args_per_parameter[par] = []
+                last_param = par
+            else:  # A numerical argument:
+                try:
+                    float(par)
+                    args_per_parameter[last_param].append(par)
+                except TypeError as err:
+                    raise TypeError(f'Invalid argument type: {par}') from err
+        return _params, args_per_parameter
+
+    def _parse_dond_arguments(*params: Union[_BaseParameter,
+                              int, float, ParamMeasT]
+                              ) -> Tuple[
+        List[_BaseParameter],
+        Dict[_BaseParameter, Dict[str, Union[float, int]]],
+        List[ParamMeasT]
+    ]:
+        """
+        Parse arguments for a dond scan.
+
+        The expected order for the arguments is:
+        params: param_set_1, start_1, stop_1, num_points_1, delay_1, ...,
+                param_set_n, start_n, stop_n, num_points_1, delay_n,
+                param_meas_1, param_meas_2, ..., param_meas_m
+        """
+        _params, args_per_parameter = _find_parameters(*params)
+        arg_types = ['start', 'stop', 'num_points', 'delay']
+        params_set: List[_BaseParameter] = []
+        params_meas: List[ParamMeasT] = []
+        params_dicts: Dict[_BaseParameter, Dict[str, Union[float, int]]] = {}
+        for param in _params:
+            arguments = args_per_parameter[param]
+            if not arguments:
+                if isinstance(param, _BaseParameter):
+                    params_meas.append(param)
+            elif len(arguments) == len(arg_types):
+                if isinstance(param, _BaseParameter):
+                    params_set.append(param)
+                    params_dicts[param] = {}
+                    for arg_type, value in zip(arg_types, arguments):
+                        params_dicts[param][arg_type] = value
+            else:
+                raise ValueError(f'Invalid number of arguments for '
+                                 'parameter {param}.')
+        return params_set, params_dicts, params_meas
+
+    params_set, params_dicts, params_meas = _parse_dond_arguments(*params)
+
+    all_setpoint_params = tuple(params_set) + tuple(
+            s for s in additional_setpoints)
+
+    measured_parameters = tuple(param for param in params_meas
```

Comment:
since `params_meas` is a list it can be converted to tuple as `tuple(params_meas)`